lib/fizzy/execute.cpp

```diff
@@ -79,6 +79,21 @@ std::vector<uint64_t> execute(const module& _module, funcidx _function, std::vec
             stack.push(a + b);
             break;
         }
+        case instr::i32_wrap_i64: {
+            const auto a = static_cast<uint64_t>(stack.pop());
+            stack.push(a & 0x00000000ffffffff);
+            break;
+        }
+        case instr::i64_extend_i32_s: {
+            const auto a = static_cast<int64_t>(stack.pop());
```

Comment:
This is wrong. You need 3 casts: `int32_t` -> `int64_t` -> `uint64_t`. I.e. you want `0x00000000ffffffff` become `0xffffffffffffffff`.